test/e2e/velero_utils.go

```diff
@@ -131,6 +139,111 @@ func installVeleroServer(io *cliinstall.InstallOptions) error {
 	return nil
 }
 
+// patch the velero resources for E2E testing
+//  1. change image pull policy as "Always"
+//  2. apply the image pull secret
+func patchResources(namespace, registryCredentialFile string, resources *unstructured.UnstructuredList) error {
+	if len(registryCredentialFile) == 0 {
+		return nil
+	}
+	credential, err := ioutil.ReadFile(registryCredentialFile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read the registry credential file %s", registryCredentialFile)
+	}
+	imagePullSecret := corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "image-pull-secret",
+			Namespace: namespace,
+		},
+		Type: corev1.SecretTypeDockerConfigJson,
+		Data: map[string][]byte{
+			".dockerconfigjson": credential,
+		},
+	}
+
+	for i, resource := range resources.Items {
```

Comment:
Please don't use i.  Give it some name, I know Go coders love this, but it's not great.